Use "import type"

# Bug report

## Describe the bug

Typescript errors when using realtime-js in SvelteKit.

`npm run check:`
```
/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/lib/push.ts:2:1
Error: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.
import { DEFAULT_TIMEOUT } from '../lib/constants'
import RealtimeChannel from '../RealtimeChannel'



/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeClient.ts:173:3
Hint: This may be converted to an async function.
   */
  removeChannel(
    channel: RealtimeChannel


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeClient.ts:187:3
Hint: This may be converted to an async function.
   */
  removeAllChannels(): Promise<RealtimeRemoveChannelResponse[]> {
    return Promise.all(


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeClient.ts:1:30
Error: Could not find a declaration file for module 'websocket'. '/Users/********/supabase-realtime-js-typescript-errors/node_modules/websocket/index.js' implicitly has an 'any' type.
  Try `npm i --save-dev @types/websocket` if it exists or add a new declaration (.d.ts) file containing `declare module 'websocket';`
import { w3cwebsocket } from 'websocket'
import {


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeClient.ts:14:27
Error: 'RealtimeChannelOptions' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.
import Serializer from './lib/serializer'
import RealtimeChannel, { RealtimeChannelOptions } from './RealtimeChannel'



/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimePresence.ts:6:1
Error: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.

import {
  PresenceOpts,
  PresenceOnJoinCallback,
  PresenceOnLeaveCallback,
} from 'phoenix'
import RealtimeChannel from './RealtimeChannel'


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimePresence.ts:7:3
Error: 'PresenceOpts' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.
import {
  PresenceOpts,
  PresenceOnJoinCallback,


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimePresence.ts:8:3
Error: 'PresenceOnJoinCallback' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.
  PresenceOpts,
  PresenceOnJoinCallback,
  PresenceOnLeaveCallback,


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimePresence.ts:9:3
Error: 'PresenceOnLeaveCallback' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.
  PresenceOnJoinCallback,
  PresenceOnLeaveCallback,
} from 'phoenix'


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimePresence.ts:11:1
Error: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.
} from 'phoenix'
import RealtimeChannel from './RealtimeChannel'



/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/lib/transformers.ts:109:20
Hint: JSDoc '@param' tag has name 'stringValue', but there is no parameter with that name.
 * @param {String} type A postgres column type
 * @param {String} stringValue The cell value
 *


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeChannel.ts:3:1
Error: This import is never used as a value and must use 'import type' because 'importsNotUsedAsValues' is set to 'error'.
import Push from './lib/push'
import RealtimeClient from './RealtimeClient'
import Timer from './lib/timer'


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeChannel.ts:6:3
Error: 'RealtimePresenceJoinPayload' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.
import RealtimePresence, {
  RealtimePresenceJoinPayload,
  RealtimePresenceLeavePayload,


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeChannel.ts:7:3
Error: 'RealtimePresenceLeavePayload' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.
  RealtimePresenceJoinPayload,
  RealtimePresenceLeavePayload,
  RealtimePresenceState,


/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimeChannel.ts:8:3
Error: 'RealtimePresenceState' is a type and must be imported using a type-only import when 'preserveValueImports' and 'isolatedModules' are both enabled.
  RealtimePresenceLeavePayload,
  RealtimePresenceState,
  REALTIME_PRESENCE_LISTEN_EVENTS,

```

`npm run dev` and visiting route that uses realtime-js
```
Cannot find module 'phoenix' imported from '/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimePresence.ts'
Error: Cannot find module 'phoenix' imported from '/Users/********/supabase-realtime-js-typescript-errors/node_modules/@supabase/realtime-js/src/RealtimePresence.ts'
    at viteResolve (file:///Users/********/supabase-realtime-js-typescript-errors/node_modules/vite/dist/node/chunks/dep-4da11a5e.js:53469:25)
    at nodeImport (file:///Users/********/supabase-realtime-js-typescript-errors/node_modules/vite/dist/node/chunks/dep-4da11a5e.js:53504:15)
    at ssrImport (file:///Users/********/supabase-realtime-js-typescript-errors/node_modules/vite/dist/node/chunks/dep-4da11a5e.js:53399:20)
    at eval (/node_modules/@supabase/realtime-js/src/RealtimePresence.ts:3:50)
    at instantiateModule (file:///Users/********/supabase-realtime-js-typescript-errors/node_modules/vite/dist/node/chunks/dep-4da11a5e.js:53445:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. `git clone https://github.com/philippviereck/supabase-realtime-js-typescript-errors.git`
2. `cd supabase-realtime-js-typescript-errors`
3. `npm i && npm run check` (see errors related to realtime-js)
4. create .env
5. `npm run dev`
6. go to `localhost:{port}/BUG` (Actual runtime error due to not using type import)

## Expected behavior

No errors.

## Screenshots

-
## System information

- OS: macOS
- Browser all
- Versions:
Example uses:
```
 "@supabase/auth-helpers-sveltekit": "^0.8.3",
"@supabase/supabase-js": "^2.0.4",
```
But it's the same for:
```
"@supabase/auth-helpers-sveltekit": "^0.9.0",
"@supabase/supabase-js": "^2.11.0",
```
- Version of Node.js: v18.14.0

--- a/package.json
+++ b/package.json
@@ -36,11 +36,11 @@
   },
   "dependencies": {
     "@types/phoenix": "^1.5.4",
+    "@types/websocket": "^1.0.3",
     "websocket": "^1.0.34"
   },
   "devDependencies": {
     "@babel/runtime": "^7.9.2",
-    "@types/websocket": "^1.0.3",
     "babel-preset-env": "^1.7.0",
     "babel-register": "^6.26.0",
     "eslint": "^7.0.0",

--- a/src/RealtimeChannel.ts
+++ b/src/RealtimeChannel.ts
@@ -3,10 +3,12 @@ import Push from './lib/push'
 import type RealtimeClient from './RealtimeClient'
 import Timer from './lib/timer'
 import RealtimePresence, {
-  type RealtimePresenceJoinPayload,
-  type RealtimePresenceLeavePayload,
-  type RealtimePresenceState,
   REALTIME_PRESENCE_LISTEN_EVENTS,
+} from './RealtimePresence'
+import type {
+  RealtimePresenceJoinPayload,
+  RealtimePresenceLeavePayload,
+  RealtimePresenceState,
 } from './RealtimePresence'
 import * as Transformers from './lib/transformers'
 

--- a/src/RealtimeChannel.ts
+++ b/src/RealtimeChannel.ts
@@ -1,11 +1,11 @@
 import { CHANNEL_EVENTS, CHANNEL_STATES } from './lib/constants'
 import Push from './lib/push'
-import RealtimeClient from './RealtimeClient'
+import type RealtimeClient from './RealtimeClient'
 import Timer from './lib/timer'
 import RealtimePresence, {
-  RealtimePresenceJoinPayload,
-  RealtimePresenceLeavePayload,
-  RealtimePresenceState,
+  type RealtimePresenceJoinPayload,
+  type RealtimePresenceLeavePayload,
+  type RealtimePresenceState,
   REALTIME_PRESENCE_LISTEN_EVENTS,
 } from './RealtimePresence'
 import * as Transformers from './lib/transformers'

--- a/src/RealtimeClient.ts
+++ b/src/RealtimeClient.ts
@@ -11,7 +11,7 @@ import {
 } from './lib/constants'
 import Timer from './lib/timer'
 import Serializer from './lib/serializer'
-import RealtimeChannel, { RealtimeChannelOptions } from './RealtimeChannel'
+import RealtimeChannel, { type RealtimeChannelOptions } from './RealtimeChannel'
 
 export type RealtimeClientOptions = {
   transport?: WebSocket

--- a/src/RealtimeClient.ts
+++ b/src/RealtimeClient.ts
@@ -170,27 +170,25 @@ export default class RealtimeClient {
    * Unsubscribes and removes a single channel
    * @param channel A RealtimeChannel instance
    */
-  removeChannel(
+  async removeChannel(
     channel: RealtimeChannel
   ): Promise<RealtimeRemoveChannelResponse> {
-    return channel.unsubscribe().then((status) => {
-      if (this.channels.length === 0) {
-        this.disconnect()
-      }
-      return status
-    })
+    const status = await channel.unsubscribe()
+    if (this.channels.length === 0) {
+      this.disconnect()
+    }
+    return status
   }
 
   /**
    * Unsubscribes and removes all channels
    */
-  removeAllChannels(): Promise<RealtimeRemoveChannelResponse[]> {
-    return Promise.all(
+  async removeAllChannels(): Promise<RealtimeRemoveChannelResponse[]> {
+    const values_1 = await Promise.all(
       this.channels.map((channel) => channel.unsubscribe())
-    ).then((values) => {
-      this.disconnect()
-      return values
-    })
+    )
+    this.disconnect()
+    return values_1
   }
 
   /**

--- a/src/RealtimeClient.ts
+++ b/src/RealtimeClient.ts
@@ -11,7 +11,8 @@ import {
 } from './lib/constants'
 import Timer from './lib/timer'
 import Serializer from './lib/serializer'
-import RealtimeChannel, { type RealtimeChannelOptions } from './RealtimeChannel'
+import RealtimeChannel from './RealtimeChannel'
+import type { RealtimeChannelOptions } from './RealtimeChannel'
 
 export type RealtimeClientOptions = {
   transport?: WebSocket

--- a/src/RealtimePresence.ts
+++ b/src/RealtimePresence.ts
@@ -3,12 +3,12 @@
   License: https://github.com/phoenixframework/phoenix/blob/d344ec0a732ab4ee204215b31de69cf4be72e3bf/LICENSE.md
 */
 
-import {
+import type {
   PresenceOpts,
   PresenceOnJoinCallback,
   PresenceOnLeaveCallback,
 } from 'phoenix'
-import RealtimeChannel from './RealtimeChannel'
+import type RealtimeChannel from './RealtimeChannel'
 
 type Presence<T extends { [key: string]: any } = {}> = {
   presence_ref: string

--- a/src/lib/push.ts
+++ b/src/lib/push.ts
@@ -1,5 +1,5 @@
 import { DEFAULT_TIMEOUT } from '../lib/constants'
-import RealtimeChannel from '../RealtimeChannel'
+import type RealtimeChannel from '../RealtimeChannel'
 
 export default class Push {
   sent: boolean = false

--- a/src/lib/transformers.ts
+++ b/src/lib/transformers.ts
@@ -106,7 +106,7 @@ export const convertColumn = (
  * If the value of the cell is `null`, returns null.
  * Otherwise converts the string value to the correct type.
  * @param {String} type A postgres column type
- * @param {String} stringValue The cell value
+ * @param {String} value The cell value
  *
  * @example convertCell('bool', 't')
  * //=> true

